x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -72,22 +75,36 @@ public static SearchableSnapshotAction parse(XContentParser parser) {
 
     private final String snapshotRepository;
     private final boolean forceMergeIndex;
+    @Nullable
+    private final Integer totalShardsPerNode;
 
-    public SearchableSnapshotAction(String snapshotRepository, boolean forceMergeIndex) {
+    public SearchableSnapshotAction(String snapshotRepository, boolean forceMergeIndex, @Nullable Integer totalShardsPerNode) {
         if (Strings.hasText(snapshotRepository) == false) {
             throw new IllegalArgumentException("the snapshot repository must be specified");
         }
         this.snapshotRepository = snapshotRepository;
         this.forceMergeIndex = forceMergeIndex;
+
+        if (totalShardsPerNode != null && totalShardsPerNode < -1) {
```

Comment:
```suggestion
        if (totalShardsPerNode != null && totalShardsPerNode < 1) {
```

(since we want to disallow -1 and 0)